beetsplug/lyrics.py

```diff
@@ -419,15 +419,23 @@ def fetch(self, artist, title):
             return None
 
         for hit in json["response"]["hits"]:
-            # Genius uses zero-width characters to denote lowercase
-            # artist names.
-            hit_artist = hit["result"]["primary_artist"]["name"]. \
-                strip(u'\u200b').lower()
+            hit_artist = self._clean(hit["result"]["primary_artist"]["name"])
 
-            if hit_artist == artist.lower():
+            if hit_artist.lower() == artist.lower():
                 return self.lyrics_from_song_page(hit["result"]["url"])
 
-        self._log.debug(u'genius: no matching artist')
+        self._log.debug(u'Genius failed to find a matching artist for \'{0}\'',
+                        artist)
+
+    def _clean(self, artist):
+        """Cleans genius-isms to help artist matching"""
+        # Genius uses zero-width characters to denote lowercase artist names
+        artist = artist.strip(u'\u200b')
+
+        # Genius uses minus-hypen compared to beets hyphen
```

Comment:
FWIW, it's more of a MusicBrainz [dash](https://en.wikipedia.org/wiki/Dash) than a beets hyphen.

Reply:
hmm, I'm using [this code](https://www.fileformat.info/info/unicode/char/2010/index.htm). Regardless, as per your other comment, I'm going to standardize them :)

Reply:
Huh, yeah, you're absolutely right about that! Never mind me. :smiley: